OpenRA.Mods.Common/Traits/TargetableFootprint.cs

```diff
@@ -0,0 +1,56 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Actor can be targeted and is a building with a footprint.")]
+	public class TargetableFootprintInfo : TargetableInfo
```

Comment:
You use `Trait<Building>`, so this dependency must be declared with a `Requires<BuildingInfo>` here.